### PR TITLE
Add view and templates for posts

### PR DESCRIPTION
**Pull request by olivier-vadiaval**
Add view for handling GET requests to author/author_id/home
Add (minimal) view for handling POST requests to author/author_id/posts
Add template for author/author_id/home
Add reusable modal template via inclusion_tags
Add reusable post bootstrap card via inclusion_tags 
author/author_id/home should show latest posts (for now) ordered by publication date.

### Some tips:
- Create super user.
- Run makemigrations and migrate. 
- Collect static files.
- Log into admin site and create an author.
- Head to http://127.0.0.1:8000/author/1/home/
- Click on (+) icon on bottom right. Add a post.
- Check other endpoints.

image

### CODE REVIEW

File: code/socialDistribution/templates/tagtemplates/modal.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                        background-color: transparent;
+                        border: none;
+                    "
+                >
+                    <span aria-hidden="true">&times;</span>
+                </button> 
+            </div>
+            <div class="modal-body">
+                {% if modal_type == "post" %}
+                {% post_form %}
```
Comment (olivier-vadiaval) [inline code/socialDistribution/templates/tagtemplates/modal.html:19]: The modal will probably be used in other places within the app. Extend this if else and modify the modal.py file to reuse this modal template.

File: code/socialDistribution/views.py [hunk 2]
```diff
@@ -1,21 +1,100 @@
...
+        else:
+            visibility = Post.PostVisibility.PUBLIC
+        
+        # temporarily set to zero; will need to fix that soon!
+        page_size = 0
+        count = 0
+
+        try:
+            post = Post.objects.create(
+                author_id=Author.objects.get(displayName='someauthor').id,  # temporary
```
Comment (tylerelston) [inline code/socialDistribution/views.py:72]: Can we make this use `get_or_create`

File: code/socialDistribution/models.py [hunk 3]
```diff
@@ -127,4 +128,50 @@ def is_public(self):
         '''
         Check if post is public
         '''
-        return self.visibility == self.PostVisibility.PUBLIC
\ No newline at end of file
+        return self.visibility == self.PostVisibility.PUBLIC
+
+    def when(self):
```
Comment (tylerelston) [inline code/socialDistribution/models.py:133]: Looks good but what do you think about using a library for this?
https://pypi.org/project/timeago/

**Review by tylerelston — CHANGES_REQUESTED** [review]
Looks great! Left a couple small comments

**Comment by olivier-vadiaval** [comment]
> @olivier-vadiaval I'll let you decide if you want to merge now and keep note of the comments or address the comments first. Thanks.

I think I'll address some of the comments and then merge.
Thanks again for the review!😃

**Pull request merged**